contracts/staking/stakeManager/StakeManager.sol

```diff
@@ -479,79 +496,198 @@ contract StakeManager is IStakeManager, StakeManagerStorage {
         reward = Math.min(CHECKPOINT_REWARD, reward);
 
         uint256 _proposerBonus = reward.mul(proposerBonus).div(100);
-        Validator storage _proposer = validators[signerToValidator[proposer]];
+        uint256 proposerId = signerToValidator[proposer];
```

Comment:
we need to move the proposer bonus to `_increaseRewardAndAssertConsensus` because proposer bonus must be deducted from the amount of the final rewards.